### PR TITLE
Update to Rust 2016-03-05

### PR DESCRIPTION
**Pull request by nox**
Blocked by https://github.com/Manishearth/rust-tenacious/pull/14.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9888)

### CODE REVIEW

**Comment by nox** [comment]
This is surprisingly way simpler than I expected it to be.

File: components/net/fetch/cors_cache.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ pub enum HeaderOrMethod {
 impl HeaderOrMethod {
     fn match_header(&self, header_name: &str) -> bool {
         match *self {
-            HeaderOrMethod::HeaderData(ref s) => s.eq_ignore_ascii_case(header_name),
+            HeaderOrMethod::HeaderData(ref s) => (&**s).eq_ignore_ascii_case(header_name),
```
Comment (Manishearth) [inline components/net/fetch/cors_cache.rs]: why is autoderef not helping us here?

Comment (nox) [inline components/net/fetch/cors_cache.rs]: https://github.com/rust-lang/rust/issues/26218

**Comment by nox** [comment]
@Manishearth Just bumped tenacious to 0.1.0 now you published it. r?

**Comment by Manishearth** [comment]
Reviewed 8 of 12 files at r1, 5 of 5 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/9888)*

**Pull request merged**